proto/semantic.proto

```diff
@@ -174,3 +178,30 @@ message Span {
   Position start = 1;
   Position end = 2;
 }
+
+message ScopeGraphFile {
+  string path = 1;
+  string language = 2;
+  repeated ScopeGraphNode nodes = 3;
+  repeated ScopeGraphPath paths = 4;
+  repeated ParseError errors = 5;
+}
+
+message ScopeGraphNode {
+  int32 id = 1;
+  string name = 2;
+  string line = 3;
+  string syntax = 4;
+  bool is_definition = 5;
+  Span span = 6;
+}
+
+message ScopeGraphPath {
+  string starting_symbol_stack = 1;
```

Comment:
These will be encoded values too, right? Colon-separated or something?

Reply:
Yes, these are currently stacks encoded and using ascii delimiters.

Reply:
Would it make more sense to have this be a `repeated string` instead?  Are we using different ascii characters as different delimiters, or as sigils at the beginning of a symbol name to indicate its type?  If the former, this is worth a comment describing the format more precisely since it becomes part of the API contract.  If the latter that would line up well with `repeated string`, and then only aleph would need to know specifically how we turn the list of symbols on the stack into a scalar string in the db.  And the per-language stack graph generators would have full flexibility to choose sigils and whatnot as needed for the language.